### PR TITLE
Add ability for marathon provider to set maxconn values, loadbalancer algorithm, and circuit breaker expression

### PR DESCRIPTION
**Pull request by jangie**
Initial implementation: Force both maxconn labels to be present to trigger behavior.

### CODE REVIEW

**Comment by jangie** [comment]
to address #612

File: provider/provider.go [hunk 1]
```diff
@@ -79,7 +80,9 @@ func (p *BaseProvider) getConfiguration(defaultTemplateFile string, funcMap temp
 		return nil, err
 	}
 
-	if _, err := toml.Decode(buffer.String(), configuration); err != nil {
+	var renderedTemplate = buffer.String()
+	log.Debugf("Rendering results of %s:\n%s", defaultTemplateFile, renderedTemplate)
```
Comment (jangie) [inline provider/provider.go]: Not sure if this is something that is desired, but it certainly helped me debug while coding this.

Comment (emilevauge) [inline provider/provider.go]: Once this PR is ready, we should comment that indeed ;)

**Comment by emilevauge** [comment]
Thanks a lot @jangie :)
LGTM :tada:

**Comment by jangie** [comment]
@emilevauge I've removed the log statement for now.

I'll be taking a stab at adding similar functionality to the other providers. Is it also your opinion that the maxconn settings should be an all-or-nothing affair? I.e., that without extractorfunc, amount should do nothing, as well as vice versa?

**Comment by emilevauge** [comment]
LGTM, but needs rebase :)

**Comment by jangie** [comment]
Rebased

**Comment by jangie** [comment]
Rebased again.

**Comment by emilevauge** [comment]
ping @containous/traefik

**Pull request merged**